### DISCUSSION TITLE
Why do created pages not show?

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3337] [Q&A] [answered]
I managed to get the admin for the CMS up and running. I created an `uncategorised` page called `home`. When I click on `View`, it takes me to a blank page. 

1. Why is it blank? (Path is `https://arc2.dev.localhost:8443/page/328/home`)
2. How can I assign it to be the home page? (Currently if I navigate to the `/` path it shows the default `That page does not exist
[Return to the homepage](https://arc2.dev.localhost:8443/)`)

**Comment by mworrell** [comment]
You can set the page_path of the page to `/`

**Comment by geoidesic** [comment]
How would I do that? 
If I try edit the page via admin, the term `page_path` appears in the `Advanced` expander.
If I set `page_path` to `/` then try reload the home page I get an error: `Load cannot follow more than 20 redirections`.

**Comment by mworrell** [comment]
(Sorry about the delay)

In the admin, the page path can be set here:

<img width="1083" alt="Screenshot 2023-04-12 at 08 32 20" src="https://user-images.githubusercontent.com/38268/231370836-67ed715a-66bd-4812-a50e-9eaf1354f73c.png">

Also, we always set the unique name of the home page to `page_home`.

If you have a template `home.tpl` (in `priv/templates`) the dispatch rule will be:

```
 {home,         [],                         controller_template,  [ {template, "home.tpl"}, {id, page_home} ]},
```

You can also debug dispatches in the admin.  Enable `mod_development` and then go to the menu System > Development, and paste the path you are debugging here:

<img width="587" alt="Screenshot 2023-04-12 at 08 35 37" src="https://user-images.githubusercontent.com/38268/231371494-3e635fe5-2d39-45e1-bad4-7ecbffa8937c.png">

It will then explain how and if the path matches and how the request will be handled.

**Comment by geoidesic** [comment]
I added `priv/home.tpl` (just `<pre>hello</pre>` for now).
I added the dispatch rule for it.
I modified the page unique name as suggested.
I no longer get the redirect error but the home page on the `/` path does not load. Instead it just says `server dropped the connection`.
Nothing in the Erlang console (i.e. no indication of a request).
I tried debugging the dispatch rules as you suggested:
<img width="1439" alt="Screenshot 2023-04-14 at 18 09 01" src="https://user-images.githubusercontent.com/4637995/232111411-50268356-ca9d-49ea-98e6-9fc2e7846dbc.png">

**Comment by mworrell** [comment]
From the dispatch I see that there is no `home` Dispatch rule matched.Did you add that rule to your dispatch file?It now tries to use the page dispatch rule (after finding your home page resource using the page-path).Regarding the dropped connection, I sometimes get that when I try a http url on the https endpoint.On 14 Apr 2023, at 19:11, Noel da Costa ***@***.***> wrote:﻿
I added priv/home.tpl (just <pre>hello</pre> for now).
I added the dispatch rule for it.
I modified the page unique name as suggested.
I no longer get the redirect error but the home page on the / path does not load. Instead it just says server dropped the connection.
Nothing in the Erlang console (i.e. no indication of a request).
I tried debugging the dispatch rules as you suggested:

**Comment by geoidesic** [comment]
How do you know that?
I think I did add the dispatch rule...
<img width="1624" alt="Screenshot 2023-04-14 at 22 48 17" src="https://user-images.githubusercontent.com/4637995/232160778-2334795d-9a1a-4bb0-88f2-5ba2ef45206d.png">
![Screenshot 2023-04-14 at 22 51 14](https://user-images.githubusercontent.com/4637995/232161112-dd720440-3bd1-4bdd-83ed-4f2899fbd0d6.png)

**Reply by mworrell** [reply]
I think I see the confusion. 

The templates and dispatch should be in your site app folder.

So something like:

`apps_user/yoursite/priv/dispatch/dispatch`

`apps_user/yoursite/priv/templates/home.tpl`

Everything that belongs to a site should be in the site's app folder.

Same for modules, which have their own app folder.

**Comment by geoidesic** [comment]
Ok tx, that seems to have worked.

1. How do I get it to output the content entered in page admin?

<img width="1624" alt="Screenshot 2023-04-14 at 23 24 43" src="https://user-images.githubusercontent.com/4637995/232164861-6a8232e2-2d28-4761-90b6-17da6849ce14.png">

2. I looked for `controller_template` file but couldn't find it anywhere?

**Reply by geoidesic** [reply]
@mworrell A year and a half later still no answer :( Presumably there's some way of referencing the CMS content from the WYSIWYG in the template?

**Reply by geoidesic** [reply, marked as answer]
Never mind, I figured it out – add this into the `home.tpl`
```
	<p>{{m.rsc[id].body}}</p>
```

**Reply by mworrell** [reply]
Ah yes, though the body can contain any HTML and also media, so best is to something like this:

```django
<div class="some-class-for-styling">
{{ id.body|show_media }}
</div>
```

The `show_media` filter ensures that images and video is show in the body text. It uses a template to render those, which makes it possible to show media differently on different pages (or in emails).

https://zotonic.com/docs/1342/show_media

The `id.body` is a convenience for `m.rsc[id].body` which works if the `id` is an integer or atom, As the template controller maps the `id` to an integer you can directly request the `body` from it, the template compiler adds the call to the `m.rsc` model.
